CMakeLists.txt

```diff
@@ -251,7 +251,12 @@ if(ENABLE_TRACY)
 endif()
 
 add_library(metal_header_directories INTERFACE)
-target_include_directories(metal_header_directories INTERFACE ${PROJECT_SOURCE_DIR}/tt_metal/hw/inc)
+target_include_directories(
+    metal_header_directories
```

Comment:
Didn't you do some work recently to get rid of these global includes?

Reply:
If there was, I must have missed that. Is there somewhere else this change should go instead?

Reply:
@tt-rkim I just put the PR up for it this morning -- it was pending a UMD bump earlier. https://github.com/tenstorrent/tt-metal/pull/14748/files

@patrickroberts I'll look over this PR and suggest what should be done instead.

Reply:
@blozano-tt and I discussed this and here's what we recommend:
* Add the API directory to the _directory scope_ in `tt_metal`
  * This provides a default for everything "inside" to freely access the API, but does not spill outside
* Add the API directory to the PUBLIC interface of the tt_metal target
  * This is what will be delivered to consumers if and only if they've stated their dependency on tt_metal

Example: https://github.com/tenstorrent/tt-metal/commit/7f8d019eea9413c07dd191848b6b1e7474414357

Disclaimer: we're treading new ground here, so we reserve the right to change our recommendation in the future if we learn better :)

Reply:
@afuller-TT Thanks for the feedback. At some point it would be nice to make the `${CMAKE_CURRENT_SOURCE_DIR}` a `PRIVATE` include directory of `tt_metal`. That would prevent headers in other directories besides `include` from being used in downstream libraries like ttnn, while enabling source files in metal library implementation to continue to include headers from anywhere under `tt_metal`, correct?
Ideally I'd also like to get rid of `${PROJECT_SOURCE_DIR}` from that list of include directories altogether, but I'm not sure how feasible that would be.

Reply:
`#include "tt_metal/*"` is sprinkled all over the code base.

It shouldn't be necessary for private header files, but getting rid of it would likely be a lot of work. (Which I support doing at some point as we divide what is public and what is private).